src/backend/distributed/transaction/citus_dist_stat_activity.c

```diff
@@ -535,18 +589,9 @@ ParseCitusDistStat(PGresult *result, int64 rowIndex)
 			cstring_to_text(initiatorWorkerNode->workerName);
 		citusDistStat->master_query_host_port = initiatorWorkerNode->workerPort;
 	}
-	else if (initiator_node_identifier == 0 && IsCoordinator())
-	{
-		citusDistStat->master_query_host_name = cstring_to_text(coordinator_host_name);
-		citusDistStat->master_query_host_port = PostPortNumber;
-	}
 	else
 	{
-		/*
-		 * We could only get here if the function is called from metadata workers and
-		 * the query is initiated from the coordinator.
-		 */
-		citusDistStat->master_query_host_name = cstring_to_text(coordinator_host_name);
+		citusDistStat->master_query_host_name = NULL;
```

Comment:
We should update the comments (I've sent an example/suggestion in the hangout chat when we talked about this change)